### ISSUE TITLE
affine grid and grid sample support in onnxruntime

### ISSUE THREAD

**Issue by stever7878** [microsoft/onnxruntime#10232]
**Environment** 
pytorch==1.9.1
onnxruntime==1.8.0
onnx==1.8.1
Python==3.6

 I am trying to convert my Pytorch model to ONNX using above environment versions. But my model needs **affine grid** and **grid sample** features to get conversion. So, I wanted to know if there is any update to add these features in upcoming release, tentative time when it will be available to the users.

**Comment by tianleiwu** [comment]
There is GridSample contrib operator in ONNX Runtime (https://github.com/microsoft/onnxruntime/pull/8551), so I think it is available to use as long as you can export your model using pytorch custom op: [example](https://github.com/microsoft/onnxruntime/blob/47435311f43cf587b40da20dc33d94bbf950e1d4/onnxruntime/python/tools/pytorch_export_contrib_ops.py#L36-L62)

GridSample was added to onnx in Nov (https://github.com/onnx/onnx/pull/3557). The operator definition will be available in onnx 1.11 release. PyTorch will need some time to support export the operator.

**Comment by stever7878** [comment]
@tianleiwu When onnx 1.11 version will be release with GridSample features ?

**Comment by tianleiwu** [comment]
@jcwchen, do you know when onnx 1.11 will be released?
Note that, after the operator spec in onnx, there still need some time (I guess a few months) to add support in ONNX Runtime and PyTorch exporter.

**Comment by jcwchen** [comment]
> @jcwchen, do you know when onnx 1.11 will be released?

ONNX will release 1.11 in the week of 2/7.

**Comment by Jove125** [comment]
> @jcwchen, do you know when onnx 1.11 will be released? Note that, after the operator spec in onnx, there still need some time (I guess a few months) to add support in ONNX Runtime and PyTorch exporter.

@tianleiwu, is there any way to do/build it manually?

**Comment by stever7878** [comment]
**Environment**
Onnx-> 1.11.0
torch -> 1.11.0+cu102
torchvision  -> 0.12.0+cu102

I am trying to convert my Pytorch model to ONNX using above environment versions. still getting same error like - 


**RuntimeError: Exporting the operator grid_sampler to ONNX opset version 11 is not supported. Please feel free to request support or submit a pull request on PyTorch GitHub.**

**Comment by PINTO0309** [comment]
I have verified that `grid_sample` works correctly in an informal way.

1. PyTorch v1.12.0-dev
2. onnxruntime v1.12.0 - opset=16

- Manually applied patches
https://github.com/pytorch/pytorch/pull/76159
![image](https://user-images.githubusercontent.com/33194443/166209532-3774ab56-ddc3-4361-b739-bad5da333456.png)

- PyTorch build
  https://zenn.dev/pinto0309/scraps/a9cf414ef289e7

- onnxruntime build
  https://zenn.dev/pinto0309/scraps/9e3b1f98ba6bd4

**Comment by imneonizer** [comment]
I have been following the issue related to grid sampler from very long, and my main interest is because I want to export STN layer  to onnx.

based on issue I found that ONNX opset 16 supports grid sample hence I followed @PINTO0309 links to build pytorch, I was able to use opset 16 however I get this issue:

````
Exporting the operator ::affine_grid_generator to ONNX opset version 16 is not supported
````

Please help me understand how do I track this issue, what onnx version and pytorch version might support it.

**Comment by stever7878** [comment]
Hello , 
@PINTO0309  Thanks for your replay !!

I can't able to find onnruntime version 1.12 . please, guide me on with link it is available.
https://github.com/microsoft/onnxruntime
Here , latest version only 1.11.1 version

**Comment by PINTO0309** [comment]
I don't know why you would need an unofficially built WHEEL, but here it is.

https://github.com/PINTO0309/openvino2tensorflow/releases

You guys just build it yourselves. The affine_grid_generator is not implemented in onnx and is not available at this time.

**Comment by Jove125** [comment]
@PINTO0309
Thanks for the answer.
I created an issue, but no one has answered yet.
Is it possible to solve the problem in another way: add this function to earlier opsets (15, 14, etc.)? If yes, how to do it?

**Comment by PINTO0309** [comment]
https://onnxruntime.ai/docs/reference/operators/add-custom-op.html

**Comment by snnn** [comment]
Resolved in #17744 .
